### DISCUSSION TITLE
Why ABP to changes the behavior of EF that convert the Datetime column to non-timezone string?

### DISCUSSION THREAD

**Discussion by tian-jie** [abpframework/abp#19637] [Q&A]
Hi,

I am using Abp 8 to build a wpf application. I see a problem that when pg installed in local machine, and CreationTime is auto generated with type "timestamp with time zone", which store datetime something like "2024-04-28 10:24:09.548593+08".

But my application is designed to deploy in different countries, which may configured as different time zones.
That might cause a problem that once PG is installed as a default time zone like "+08", and user changed the windows system time zone to "-04", that means the time zone are different between database and application.

I tracked the database log, the data creation sql for the datetime type is passed like "2024-04-28 10:24:09.548593", which does not contains a time zone, while I also tried for a pure .net core web MVC application with EF, the data passed as a datetime contains time zone field.
![image](https://github.com/abpframework/abp/assets/25471485/a7bf6337-7275-4f79-8ff3-13697e146d5b)

You know, if I can pass the datetime string with time zone field, that should be not a problem.

My question is, why we removed the time zone field in this case, what is the way to default add the time zone field to the datetime string passed to EF in abp wpf application?

**Comment by maliming** [comment]
hi

Please check the document https://docs.abp.io/en/abp/latest/Timing#datetime-normalization

**Comment by tian-jie** [comment]
Yes Ma, I went through that document. 

My problem is that I hope I can send as following: "2024-04-28 10:24:09.548593+08" or "2024-04-28 10:24:09.548593z"

is there any way to add the time zone field passed to SQL?

**Reply by tian-jie** [reply]
Currently I am using following code to change the column type from "timestamp with time zone" to "timestamp **without** time zone", and make a rule that code to database should save UTC time.
``` csharp
System.AppContext.SetSwitch("Npgsql.EnableLegacyTimestampBehavior", true);
```

But I still would like to understand why the time zone field is missing, and how I can add it back if better.

**Reply by amerina** [reply]
if you has many time zone,the best may be use DateTime.ToUniversalTime to store UTC Time

**Reply by amerina** [reply]
> Yes Ma, I went through that document.
> 
> My problem is that I hope I can send as following: "2024-04-28 10:24:09.548593+08" or "2024-04-28 10:24:09.548593z"
> 
> is there any way to add the time zone field passed to SQL?

May be you can try DateTimeOffset type
